File: 05-04-2024 Strictly Increasing Array/Solution.py
class Solution:
    def min_operations(self,nums):
        # Code here
        dp = [1 for i in range(len(nums))]
        dp[0] = 1
        for i in range(1, len(nums)):
            for j in range(i):
                if nums[i] > nums[j] and nums[i] - nums[j] >= (i-j):
                    dp[i] = max(dp[i], 1 + dp[j])
        max_ls = max(dp)
        return len(nums) - max_ls

File: 05-04-2024 Strictly Increasing Array/Solution.java
class Solution
{
    public int solve(int []nums,int prev,int index)
    {
        if(index==nums.length)
        {
            return 0;
        }
        if(prev==-1)
        {   
            return Math.min(solve(nums,nums[index],index+1),1+solve(nums,-22,index+1));    
        }
        else if(nums[index]<=prev)
        {
          return 1+solve(nums,prev+1,index+1);
        }
        else
        {
            return Math.min(solve(nums,nums[index],index+1),1+solve(nums,prev+1,index+1));
        }
        
    }
    
    public int min_operations(int []nums)
    {
        // Code here
        return solve(nums,-1,0);
    }
}